test/models/patient_notification_eligibility_test.rb

```diff
@@ -0,0 +1,650 @@
+# frozen_string_literal: true
+
+require 'test_case'
+
+class PatientNotificationEligibilityTest < ActiveSupport::TestCase
+  def setup
+    # Assume that the default patient created here will have a nil preferred_contact_time
+    # and if it will be non-nil, then the specific test can change the Timecop time.
+    # Default timezone is Eastern Time.
+    Timecop.freeze(Time.now.in_time_zone('Eastern Time (US & Canada)').noon.utc)
+    @original_reporting_period = ADMIN_OPTIONS['reporting_period_minutes']
+  end
+
+  def teardown
+    ADMIN_OPTIONS['reporting_period_minutes'] = @original_reporting_period
+    Timecop.return
+  end
+
+  def default_days_ago(days)
+    (Time.now.in_time_zone('Eastern Time (US & Canada)') - days.days)
+  end
+
+  def expected_eligibility(patient, exp_eligibility)
+    assert_eligible(patient) if exp_eligibility
+    assert_ineligible(patient) unless exp_eligibility
+  end
+
+  def assert_eligible(patient)
+    scope_eligible = !Patient.better_reminder_eligible.find_by(id: patient.id).nil?
+    puts patient.report_eligibility unless scope_eligible
+    puts "\nFailing scope eligible test with: #{format_patient_str(patient)}" unless scope_eligible
+    assert scope_eligible
+
```

Comment:
Why are these commented out (and in the test below)?

Reply:
Commented out because we determined that the `report_eligibility` method should not have exact parity with the `reminder_eligible` scope. I'll remove the commented out code.